### PR TITLE
feat: Product inventory availability

### PR DESCRIPTION
**Pull request by ynnoj**
Adds `X in stock`, `Out of stock` badges to product page.

### CODE REVIEW

File: src/shopkit/index.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
 import React, { createContext } from 'react'
-import { createClient } from '@moltin/request'
+import { MoltinClient } from '@moltin/request'
 
 import { CartProvider, Cartkit } from './Cartkit'
 import { CustomerProvider, Customerkit } from './Customerkit'
 import { CheckoutProvider, Checkoutkit } from './Checkoutkit'
 
+class MoltinNodeStorageAdapter {
```
Comment (notrab) [inline src/shopkit/index.js]: @ynnoj can we rename this to `MoltinLocalStorageAdapter` instead?

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "private": true,
   "dependencies": {
     "@moltin/gatsby-source-moltin": "1.2.0",
-    "@moltin/request": "1.4.4",
+    "@moltin/request": "^2.0.2",
```
Comment (notrab) [inline package.json]:
```suggestion
    "@moltin/request": "2.0.2",
```

File: src/templates/ProductPage.js [hunk 3]
```diff
@@ -1,16 +1,41 @@
...
+import { Shopkit } from '../shopkit'
 
 function ProductPage({ data: { product } }) {
+  const { moltin } = useContext(Shopkit)
+  const [inventoryAvailability, setInventoryAvailability] = useState(0)
+  const [inventoryLoading, setInventoryLoading] = useState(true)
+  const [inventoryError, setInventoryError] = useState(false)
+
+  useEffect(() => {
+    async function getProductInventory() {
```
Comment (notrab) [inline src/templates/ProductPage.js]: @ynnoj maybe worth moving the function declaration outside of `useEffect`.

File: src/templates/ProductPage.js [hunk 4]
```diff
@@ -1,16 +1,41 @@
...
+      try {
+        const {
+          data: { available }
+        } = await moltin.get(`inventories/${product.id}`)
+
+        setInventoryLoading(false)
+        setInventoryAvailability(available)
```
Comment (notrab) [inline src/templates/ProductPage.js]: @ynnoj what if `manage_stock` is set to false?

Comment (ynnoj) [reply src/templates/ProductPage.js]: Does that not create an associated inventory record?

Comment (notrab) [reply src/templates/ProductPage.js]: https://deploy-preview-18--gatsby-demo-store.netlify.com/products/crown has `manage_stock` set to `false`, yet it still shows as out of stock and prevents me from adding to cart.

File: src/templates/ProductPage.js [hunk 5]
```diff
@@ -1,16 +1,41 @@
...
+      } catch (error) {
+        setInventoryError(error)
```
Comment (notrab) [inline src/templates/ProductPage.js]: I'm unsure how this will display the error further down. As `errors` will be an array.

Comment (notrab) [reply src/templates/ProductPage.js]: Also I get an `uncaught Promise` error when trying to add more items than I have in stock.

File: src/shopkit/index.js (hunk 1 above)
Comment (ynnoj) [reply src/shopkit/index.js]: @notrab Forgot to commit that change. Probably worth abstracting to a separate package as noted in commit message.

File: src/templates/ProductPage.js (hunk 4 above)
Comment (ynnoj) [reply src/templates/ProductPage.js]: I thought if `manage_stock` was set to false on creation then no inventory record was created. In this instance, there is an inventory record.

I guess we'll have to check `manage_stock` on the product.

File: src/templates/ProductPage.js (hunk 5 above)
Comment (ynnoj) [reply src/templates/ProductPage.js]: > Also I get an `uncaught Promise` error when trying to add more items than I have in stock.

I don't see this. I see a `201`, which is probably even more concerning.

File: src/templates/ProductPage.js (hunk 4 above)
Comment (notrab) [reply src/templates/ProductPage.js]: What if I set it to false because I no longer wish to manage stock?

**Pull request merged**